### PR TITLE
Course search: fix routing, search clearing, add search label and fix SQL queries

### PR DESCRIPTION
**Pull request by regisb**

### CODE REVIEW

**Comment by regisb** [comment]
Come on Jenkins, test this please.

**Comment by regisb** [comment]
Come on Jenkins, test this please.

**Comment by regisb** [comment]
Jenkins test this please.

File: course_pages/templates/course_pages/index.html [hunk 1]
```diff
@@ -478,14 +506,21 @@
...
+            },
             render: function() {
-                /* add some fake blocks to get correct alignment on last line */
-                this.$el.find('.fake').remove();  // clear previousely added fake blocks
+                // Remove spinner
+                this.$el.find('.syncing').remove();
+                // clear previously added fake blocks
+                this.$el.find('.fake').remove();
+                // add some fake blocks to get correct alignment on last line
                 for (var i = 5; i >= 0; i--) {
```
Comment (rmoch) [inline course_pages/templates/course_pages/index.html]: :+1:

**Comment by rmoch** [comment]
lgtm

**Pull request closed without merging**